js/huobi.js

```diff
@@ -4918,36 +4922,70 @@ module.exports = class huobi extends Exchange {
-            const accountsByType = this.safeValue (this.options, 'accountsByType', {});
-            fromAccount = fromAccount.toLowerCase (); // pro, futures
-            toAccount = toAccount.toLowerCase (); // pro, futures
-            const fromId = this.safeString (accountsByType, fromAccount, fromAccount);
-            const toId = this.safeString (accountsByType, toAccount, toAccount);
-            type = fromId + '-to-' + toId;
-        }
+        fromAccount = fromAccount.toLowerCase ();
+        toAccount = toAccount.toLowerCase ();
         const request = {
             'currency': currency['id'],
             'amount': parseFloat (this.currencyToPrecision (code, amount)),
-            'type': type,
         };
-        const response = await this.spotPrivatePostFuturesTransfer (this.extend (request, params));
+        if (fromAccount === 'swap' || toAccount === 'swap') {
+            const defaultSubType = this.safeStringLower2 (this.options, 'defaultSubType', 'subType');
+            const subType = this.safeStringLower (params, 'subType', defaultSubType);
+            if (subType === 'linear') {
+                if (fromAccount === 'swap') {
+                    fromAccount = 'linear-swap';
+                } else {
+                    toAccount = 'linear-swap';
+                }
+                const marginAccount = this.safeStringUpper2 (params, 'marginAccount', 'margin-account');
+                if (marginAccount === undefined) {
+                    throw new ArgumentsRequired (this.id + ' requires params["marginAccount"] when type is swap and subType is linear');
+                } else if (marginAccount === 'CROSS' || marginAccount === 'USDT') {
+                    request['margin-account'] = 'USDT';
+                } else {
+                    const market = this.market (marginAccount);
+                    request['margin-account'] = market['id'];
+                }
+            }
+            params = this.omit (params, [ 'subType', 'marginAccount' ]);
+        }
+        fromAccount = (fromAccount === 'future') ? 'futures' : fromAccount;
+        toAccount = (toAccount === 'future') ? 'futures' : toAccount;
+        request['from'] = fromAccount;
+        request['to'] = toAccount;
+        const response = await this.v2PrivatePostAccountTransfer (this.extend (request, params));
         //
-        //     {
-        //         "data": 12345,
-        //         "status": "ok"
-        //     }
+        //    {
+        //        code: '200',
+        //        data: '650969068',
+        //        message: 'Succeed',
+        //        success: true,
+        //        'print-log': true
+        //    }
         //
+        const success = this.safeValue (response, 'success');
+        if (!success) {
+            const message = this.safeString (response, 'message');
+            const code = this.safeString (response, 'code');
+            const ErrorType = this.safeValue (this.exceptions['exact'], code);
```

Comment:
@kroitor @frosty00 I wasn't quite sure if this was done correctly

Reply:
@carlosmiei What's your opinion?

Reply:
@samgermain I don't think this is the right place to do it, why not handle this inside `handleErrors` (and make use throwExactlyMatchedException/throwBroadlyMatchedException)? 

Every response goes through `handleErrors` before reaching our unified methods, so we usually assume that if we're parsing it here, it is a "successful" one, otherwise, we would have thrown it before

Reply:
The reason I handle the error here is because `handleErrors` was not being called

Reply:
In the most recent commit I call `throwExactlyMatchedException` followed by `throw new Error` in case no exception is thrown from `throwExactlyMatchedException`. If the most recent commit is reverted, then  `throwBroadlyMatchedException` is called instead of `throw new Error`. I didn't really understand `throwBroadlyMatchedException`, because there are so few items inside `huobi.exceptions['broad']`, so it seems like most of the time, an exception will not be thrown

Reply:
@samgermain 

> the reason I handle the error here is because handleErrors was not being called
can you please double-check this? HandleErrors is called automatically for each request. you can see in the image



![image](https://user-images.githubusercontent.com/43336371/186698584-597bf427-7a0c-46a3-af52-883766a83afb.png)

Reply:
> @samgermain
> 
> > the reason I handle the error here is because handleErrors was not being called
> 
> can you please double-check this? HandleErrors is called automatically for each request. you can see in the image
> 
> ![image](https://user-images.githubusercontent.com/43336371/186698584-597bf427-7a0c-46a3-af52-883766a83afb.png)

It's called prior to sending the request to the exchange